### ISSUE TITLE
stops working after some time

### ISSUE THREAD

**Issue by Ragnarre** [jdeath/Roblox-Homeassistant#5]
user status goes "unknown" and loses avatar picture at least once in a day, home assistant restart brings back real status. settings are default, even update interval is 2 minutes. what could cause that?

**Comment by jdeath** [comment]
I had same behavior too, after seeing your issue. Maybe the API changed or got rate limited. I will keep watching it.

**Comment by Ragnarre** [comment]
Is there any workaround to restart this component without needing to restart homeassistant?

**Comment by jdeath** [comment]
I think their API has some issues causing the integration to fail. My integration is pretty basic and has limited error checking.

The update function has a try/except, so it should not die. Each update does a fresh login. I am not sure why it does not recover. Will require someone more experienced than me to figure out.

**Comment by mkstalen** [comment]
Yeah. I've just tried installing this and it's not working at all for me.
I just get a return of "unknown" and no avatar pictures.
![image](https://github.com/user-attachments/assets/e87601ae-d8bc-4ad1-8771-cfe845e4cfea)

**Comment by jdeath** [comment]
yup. Seems not to work (family does not play much anymore). I'll take a look

**Comment by jdeath** [comment]
v1.8 should fix this. The UI will now show the ID instead of the username. The API removed the call I used to get the username for the ID. I will figure out a better method in the future, but this will work for now. I'll leave issue open until I can fix it better.

**Comment by jdeath** [comment]
fixed in v1.9
